ui/components/app/snaps/snap-ui-address-input/snap-ui-address-input.tsx

```diff
@@ -196,11 +211,13 @@ export const SnapUIAddressInput: FunctionComponent<
   const handleBlur = () => setCurrentFocusedInput(null);
 
   const handleClear = () => {
-    setValue('');
-    handleInputChange(name, '', form);
+    if (!disabled) {
+      setValue('');
+      handleInputChange(name, '', form);
+    }
   };
 
-  if (displayName) {
+  if (displayName && !error) {
```

Comment:
Why do we not want to show the matched name if there's an error?

Reply:
When something is matched imo it represents a successful operation, allowing the matched component to render when there's an error present feels counterintuitive.

Reply:
My issue with this is that we could match an address before the snap sends the error meaning that for a short time we will display the matched address and then remove it due to the error :thinking:

Reply:
I think both should be shown regardless tbh.

Reply:
Hmm yeah it's almost impossible to time that, maybe it's easier to just display regardless.